src/include/duckdb/main/appender.hpp

```diff
@@ -118,8 +125,12 @@ class Appender : public BaseAppender {
 	shared_ptr<ClientContext> context;
 	//! The table description (including column names)
 	unique_ptr<TableDescription> description;
-	//! The default expressions
-	unordered_map<idx_t, Value> default_values;
+
+	//! The default expressions.
+	unordered_map<column_t, Value> default_values;
+	//! If not empty, then this holds all columns provided by the appender.
+	//! Any other columns default to NULL, or their default values.
+	case_insensitive_set_t active_columns;
```

Comment:
I think this should be a `vector`, not a `set`. The order in which we specify columns can also be switched around in a regular insert, e.g.:

```sql
INSERT INTO tbl (col1, col2) VALUES ...;
INSERT INTO tbl (col2, col1) VALUES ...;
```